atlp-rwanda/champs-ec-be
# #187300241-Implement Google OAuth2 login functionality for streamlined authentication

**What does this PR do?**

This PR implements the necessary tasks to enable users to log in with Google OAuth2.

**Description of Task to be completed:**

 .Implemented route for handling Google OAuth2 callback.
 .Defined serialization and deserialization methods for Passport.
 .Created route to initiate Google OAuth2 login.
 .Returned user data and token in response after successful authentication.
 .Enhanced user experience by enabling login via Google OAuth2.

**How should this be manually tested?**

.Test the implementation by initiating the Google OAuth2 login process.
.Verify that users can successfully log in using their Google accounts.
.Ensure that user data and tokens are returned correctly upon successful authentication.
.Manually inspect the user experience to confirm the integration of Google OAuth2 login.

**.What are the relevant pivotal tracker/Trello stories?**

Pivotal ID #187300241
